lucky-nums/app.py

```diff
@@ -7,4 +9,29 @@
 def homepage():
     """Show homepage."""
 
-    return render_template("index.html")
\ No newline at end of file
+    return render_template("index.html")
+
+@app.route("/api/get-lucky-num", methods=['POST'])
+def lucky_num():
+    """POST route to process form data and send json response"""
+
+    name = request.json["name"]
+    email = request.json["email"]
+    year = request.json["year"]
+    color = request.json["color"]
+
+    req = {
+        "name": name,
+        "email": email,
+        "year": year,
+        "color": color
+    }
+
+    validation = validate_request(req)
+
+    if validation == True:
+        num = random.randint(1,100)
+        data = get_num_facts(num, year)
+        return make_response(data, 200)
+
+    return make_response(validation, 200)
```

Comment:
Following exception handling in my `validate_request` comment
```suggestion
    try:
        validation = validate_request(req)

        num = random.randint(1,100)
        data = get_num_facts(num, year)
        return make_response(data, 200)
    except ValidationError as e:
        return make_response(e.errors, 422)
```